### ISSUE TITLE
Show dependencies before deleting schemas and tables

### ISSUE THREAD

**Issue by kgodey** [mathesar-foundation/mathesar#399]
## Problem

We should show a list or summary of dependent objects before deleting a schema or a table.

## Proposed solution

Use the new dependencies APIs to get a list of objects and show it in the error message.

## Additional context

- Blocked by #398
- ["Working with Schemas" design spec](https://wiki.mathesar.org/design/specs/schemas)
- ["Basic Table Operations" design spec](https://wiki.mathesar.org/en/design/specs/table-operations)

**Comment by rajatvijay** [comment]
@kgodey marking this as `draft` since the `blocked by` issue is closed.

**Comment by kgodey** [comment]
Closing this, too old, requirements are likely out of date. We can create a new issue if we need this functionality in the future.
